test/ng/ngPropSpec.js

```diff
@@ -0,0 +1,775 @@
+        var element = $compile('<div ng-prop-inner_h_t_m_l="html"></div>')($rootScope);
+        $rootScope.html = $sce.trustAsHtml('hello');
+        $rootScope.$digest();
+        expect(lowercase(element.html())).toEqual('hello');
+        $rootScope.html = $sce.trustAsHtml('goodbye');
+        $rootScope.$digest();
+        expect(lowercase(element.html())).toEqual('goodbye');
+      }));
+
+      it('should not cause infinite recursion for trustAsHtml object watches',
+          inject(function($rootScope, $compile, $sce) {
+        // Ref: https://github.com/angular/angular.js/issues/3932
+        // If the binding is a function that creates a new value on every call via trustAs, we'll
+        // trigger an infinite digest if we don't take care of it.
+        var element = $compile('<div ng-prop-inner_h_t_m_l="getHtml()"></div>')($rootScope);
+        $rootScope.getHtml = function() {
+          return $sce.trustAsHtml('<div onclick="">hello</div>');
+        };
+        $rootScope.$digest();
+        expect(lowercase(element.html())).toEqual('<div onclick="">hello</div>');
+      }));
+
+      it('should handle custom $sce objects', function() {
+        function MySafeHtml(val) { this.val = val; }
+
+        module(function($provide) {
+          $provide.decorator('$sce', function($delegate) {
+            $delegate.trustAsHtml = function(html) { return new MySafeHtml(html); };
+            $delegate.getTrusted = function(type, mySafeHtml) { return mySafeHtml && mySafeHtml.val; };
+            $delegate.valueOf = function(v) { return v instanceof MySafeHtml ? v.val : v; };
+            return $delegate;
+          });
+        });
+
+        inject(function($rootScope, $compile, $sce) {
+          // Ref: https://github.com/angular/angular.js/issues/14526
+          // Previous code used toString for change detection, which fails for custom objects
+          // that don't override toString.
+          var element = $compile('<div ng-prop-inner_h_t_m_l="getHtml()"></div>')($rootScope);
+          var html = 'hello';
+          $rootScope.getHtml = function() { return $sce.trustAsHtml(html); };
+          $rootScope.$digest();
+          expect(lowercase(element.html())).toEqual('hello');
+          html = 'goodbye';
+          $rootScope.$digest();
+          expect(lowercase(element.html())).toEqual('goodbye');
+        });
+      });
+
+      describe('when $sanitize is available', function() {
+        beforeEach(function() { module('ngSanitize'); });
+
+        it('should sanitize untrusted html', inject(function($rootScope, $compile) {
+          var element = $compile('<div ng-prop-inner_h_t_m_l="html"></div>')($rootScope);
+          $rootScope.html = '<div onclick="">hello</div>';
+          $rootScope.$digest();
+          expect(lowercase(element.html())).toEqual('<div>hello</div>');
+        }));
+      });
+    });
+
+  });
+
+  describe('*[style]', function() {
```

Comment:
I think these tests are actually all invalid because you can't assign to `HTMLElement.style`. The attribute version can be assigned as a plain string, but the property is of type `CSSStyleDeclaration` and can't be assigned.

The tests are failing in some browsers which seem to throw when assigning to `HTMLElement.style`.

Should we have any special handling for non-assignable properties like this? Should we drop the `*|style => $sce.CSS` declaration?

Reply:
Although some browsers seem to accept string assignments, so maybe these tests just need an `if (canAssignToStyleProp)` around them...

Reply:
Anyone have suggestions for how to do the `canAssignToStyleProp`? Only thing I can think of is:
```
  function canAssingStyleProp() {
    try {
      document.createElement("div").style = "background: red";
      return true;
    } catch (e) {
      return false;
    }
  }
```
... however I'd rather not depend on something like that in tests.

Reply:
Why not? (BTW, there are browsers (e.g. IE11) that don't throw an error but ignore the assignment.)

Reply:
> Why not?

Because I don't trust my feature-test in tests, unless we tested them. But if we can't think of anything else I'll do it. Might use the `msie` variable for the IE detection though...

Reply:
Do you know which browsers are problematic?

Reply:
I think https://travis-ci.org/angular/angular.js/jobs/406066641#L771 has a good summary, but it's hard to tell which failures are for which browser. Is there a way of seeing that? Logs somewhere?

But you can see the following errors:
* Expected '0px' to equal '10px'.
* TypeError: Attempted to assign to readonly property. (line 7653)
* Expected '' to equal '10px'

(there's also an undefined var in there which I'll push a fix for in a sec...)

Reply:
Weird that the browser reporting the error is not shown. Maybe our reporter is outdated?
I don't know of any other way to find it. But I would be fine adding a `canAssingStyleProp()` check fwiw :grin:

Reply:
I've added 08898499df2c393e724cacf7b26f887e510ba911, but I'm taking a guess at which browser does `''` vs `'0px'` (and I'm probably wrong now that I think about it more) and put `// Support: ??` for the case where it throws.

Reply:
My guess was correct!

Just one odd IE9 error now... https://travis-ci.org/angular/angular.js/jobs/406368530#L687